### PR TITLE
Add back tagger data augmentation + Fixes for analyze_errors.py

### PR DESCRIPTION
**Pull request by laituan245**
Changes:

- Add back tagger data augmentation (but default to be False)
- Fixes for analyze_errors.py
- Add duplex tn to README

@yzhang123

### CODE REVIEW

File: docs/source/nlp/text_normalization.rst [hunk 1]
```diff
@@ -130,9 +130,21 @@ then we will simply append the prefix ``tn`` to it and so the final input to our
 be ``tn I live in tn 123 King Ave``. Similarly, for the ITN problem, we just append the prefix ``itn``
 to the input.
 
-To improve the robustness of the decoder, we also apply a simple data augmentation technique during training the decoder.
+To improve the effectiveness and robustness of our models, we also experiment with some simple data
+augmentation techniques during training.
 
-Data Augmentation for Training DuplexDecoderModel
+Data Augmentation for Training DuplexTaggerModel (Set to be False by defalt)
```
Comment (yzhang123) [inline docs/source/nlp/text_normalization.rst:136]: typo

File: docs/source/nlp/text_normalization.rst [hunk 2]
```diff
@@ -130,9 +130,21 @@ then we will simply append the prefix ``tn`` to it and so the final input to our
...
+In the Google English TN training data, about 93% of the tokens are not in any semiotic span. In other words, the ground-truth tags of most tokens are of trivial types (i.e., ``SAME`` and ``PUNCT``). To alleviate this class imbalance problem,
+for each original instance with several semiotic spans, we create a new instance by simply concatenating all the semiotic spans together. For example, considering the following ITN instance:
+
+Original instance: ``[The|SAME] [revenues|SAME] [grew|SAME] [a|SAME] [lot|SAME] [between|SAME] [two|B-TRANSFORM] [thousand|I-TRANSFORM] [two|I-TRANSFORM] [and|SAME] [two|B-TRANSFORM] [thousand|I-TRANSFORM] [five|I-TRANSFORM] [.|PUNCT]``
+
+Augmented instance: ``[two|B-TRANSFORM] [thousand|I-TRANSFORM] [two|I-TRANSFORM] [two|B-TRANSFORM] [thousand|I-TRANSFORM] [five|I-TRANSFORM]``
+
+The argument ``data.train_ds.tagger_data_augmentation`` in the config file controls whether this data augmentation will be enabled or not.
+
+Data Augmentation for Training DuplexDecoderModel (Set to be True by defalt)
```
Comment (yzhang123) [inline docs/source/nlp/text_normalization.rst:147]: typo

File: examples/nlp/duplex_text_normalization/analyze_errors.py [hunk 3]
```diff
@@ -284,8 +285,8 @@ def analyze(errors_log_fp: str, visualization_fp: str):
 if __name__ == '__main__':
     # Parse argument
     parser = ArgumentParser()
-    parser.add_argument('--errors_log_fp', type='str', help='Path to the error log file', required=True)
-    parser.add_argument('--visualization_fp', type='str', help='Path to the output visualization file', required=True)
+    parser.add_argument('--errors_log_fp', help='Path to the error log file', required=True)
+    parser.add_argument('--visualization_fp', help='Path to the output visualization file', required=True)
```
Comment (yzhang123) [inline examples/nlp/duplex_text_normalization/analyze_errors.py:289]: I still get error:
  File "analyze_errors.py", line 160, in from_lines
    return cls(_input, target, pred, mode)
UnboundLocalError: local variable 'pred' referenced before assignment

**Review by yzhang123 — CHANGES_REQUESTED** [review]

File: docs/source/nlp/text_normalization.rst (hunk 1 above)
Comment (laituan245) [reply docs/source/nlp/text_normalization.rst]: Fixed

File: docs/source/nlp/text_normalization.rst (hunk 2 above)
Comment (laituan245) [reply docs/source/nlp/text_normalization.rst]: Fixed

**Review by yzhang123 — APPROVED** [review]

**Pull request merged**